libs/pilight/protocols/433.92/hideki.c

```diff
@@ -0,0 +1,497 @@
+	if(channel >= 5) channel -= 1;
+	rc = packet[1] & 0x0F;	// rolling code
+	temp = (packet[5] & 0x0F) * 100 + ((packet[4] & 0xF0) >> 4) * 10 + (packet[4] & 0x0F);
+	if(((packet[5]>>7) & 0x01) == 0) {
+		temp = -temp;
+	}
+
+	battery_ok = (packet[5]>>6) & 0x01;
+#ifdef HIDEKI_DEBUG
+	id = packet[3]; // probably some ID
+#endif
+
+#ifdef HIDEKI_DEBUG
+	logprintf(LOG_DEBUG, "HIDEKI parseCode(): channel %02X, ID:  %02X, RC: %02X, temp: %02X, batt: %02X, sensortyp: %02x, hum: %02X",
+		  channel, id, rc, temp, battery_ok, sensortype, ((packet[6] & 0xF0) >> 4) * 10 + (packet[6] & 0x0F));
+#endif
+
+	// read some settings
+	struct settings_t *my_settings = get_settings((double)channel);
+	if(my_settings) {
+		humi_offset = my_settings->humi;
+ 		temp_offset = my_settings->temp;
+ 		wind_dir_offset = my_settings->wind_dir;
+ 		wind_factor = my_settings->wind_factor;
+ 		rain_factor = my_settings->rain;
+ 		
+		if((battery_ok == 0) && (my_settings->ignore_temp)) {
+			ignore_temperature = 1; // workaround for case that battery=0: temperature value might be invalid in that case
+		}
+ 		
+		if(my_settings->calibration > 0) {
+			/* still in calibration phase to memorize correct sensor type */
+#ifdef HIDEKI_DEBUG
+			logprintf(LOG_DEBUG, "HIDEKI parseCode(): calibration: %d, type: %d -> %d",
+				  my_settings->calibration, my_settings->sensortype, sensortype);
+#endif
+			my_settings->calibration--;
+			if(my_settings->sensortype == HIDEKI_UNKNOWN) {
+				my_settings->sensortype = sensortype;
+			} else if(my_settings->sensortype != sensortype) {
+ 				my_settings->sensortype = sensortype;
+ 				my_settings->calibration = CALIBRATION_COUNT;
+ 			}
+ 		} else if(my_settings->sensortype != sensortype) {
+			/* calibration is finished and detected sensor type does not match */
+ 			logprintf(LOG_NOTICE, "HIDEKI parseCode(): wrong sensor type found: %d (expected: %d) -> ignoring data",
+				  sensortype, my_settings->sensortype, sensortype);
+			return;	// ignore the data
+ 		}
+	} else {
+		/* no valid settings/configuration found -> the received data do not belong to a configured device from config.json
+		 * -> usually we could return here, but in that case the tool "pilight_receive" might be unable to receive other data
+		 * than locally configured... */
+  	}
+	
+	if(sensortype == HIDEKI_TS04) {
+		temperature = (double)temp/10 + temp_offset;
+		humidity = ((packet[6] & 0xF0) >> 4) * 10 + (packet[6] & 0x0F) + humi_offset;
+
+		hideki->message = json_mkobject();
+		if(ignore_temperature == 0) {
+			json_append_member(hideki->message, "temperature", json_mknumber(temperature, 1));
+		}
+		json_append_member(hideki->message, "humidity", json_mknumber(humidity, 0));
+
+		json_append_member(hideki->message, "battery", json_mknumber(battery_ok, 0));
+		json_append_member(hideki->message, "channel", json_mknumber(channel, 0));
+		json_append_member(hideki->message, "rc", json_mknumber(rc, 0));
+      } else if(sensortype == HIDEKI_WIND) {
```

Comment:
You're pretty inconsistent in using spaces and tabs. That's why the indenting is wrong here using a tabular view.

Reply:
I don't use an editor which shows the differences between spaces and tabs, so you are right. For me personally, this is not a problem.

Reply:
I use notepad++ with a tabs = 2 spaces. And indeed it anoys me :smile:.

Reply:
The other `sensortype` conditions also contain spaces.